api/src/lib.rs

```diff
@@ -304,6 +310,82 @@ where
     Ok(StatusCode::Created)
 }
 
+async fn verify_verification_token<T>(
+    mut req: Request<State<T>>,
+) -> Result<StatusCode>
+where
+    T: Database,
+{
+    #[derive(Deserialize)]
+    struct VerifyVerifyBody
+    {
+        secret: String,
+        email: String,
+    }
+
+    let body: VerifyVerifyBody = req.body_json().await.map_err(server_err)?;
+
+    let db = &req.state().db;
+    let id = &req.ext::<VaultId>().unwrap().0;
+
+    verify_token::verify(db, id, &body.secret, VerifyMethod::Email(body.email))
+        .await
+        .map_err(server_err)?;
+
+    Ok(StatusCode::Ok)
+}
+
+async fn get_verification_status<T>(mut req: Request<State<T>>) -> Result
+where
+    T: Database,
+{
+    let db = &req.state().db;
+    let id = &req.ext::<VaultId>().unwrap().0;
+
+    let mut response = Response::builder(StatusCode::Ok)
+        .header("content-type", "application/json");
+
+    match verify_token::get(db, id).await.map_err(server_err) {
```

Comment:
missing resource in this case means the user has yet to try to verify their identity, we can't distinguish that from a case where they started the process, since we don't create the verification token when the vault is created (we could, but we don't)